src/Net35/ConcurrentDictionary.cs

```diff
@@ -28,14 +28,6 @@ public ConcurrentDictionary()
             this.syncRoot = new object();
         }
 
-        public new bool TryGetValue(TKey key, out TValue value)
```

Comment:
This method is needed so TryGetValue and GetOrAdd are synchronized; otherwise get may throw exceptions when add is changing the internal data structure of the dictionary. The issue is that inside the lock it should have called `base.TryGetValue(key, out value)`. In GetOrAdd it is also better to call the base's methods.